pkg/registry/storage/csidriver/strategy.go

```diff
@@ -82,31 +83,33 @@ func (csiDriverStrategy) AllowCreateOnUpdate() bool {
 // existing object does not already have that field set. This allows the field to remain when
 // downgrading to a version that has the feature disabled.
 func (csiDriverStrategy) PrepareForUpdate(ctx context.Context, obj, old runtime.Object) {
-	if old.(*storage.CSIDriver).Spec.StorageCapacity == nil &&
+	newCSIDriver := obj.(*storage.CSIDriver)
+	oldCSIDriver := old.(*storage.CSIDriver)
+
+	if oldCSIDriver.Spec.StorageCapacity == nil &&
 		!utilfeature.DefaultFeatureGate.Enabled(features.CSIStorageCapacity) {
-		newCSIDriver := obj.(*storage.CSIDriver)
 		newCSIDriver.Spec.StorageCapacity = nil
 	}
-	if old.(*storage.CSIDriver).Spec.VolumeLifecycleModes == nil &&
+	if oldCSIDriver.Spec.VolumeLifecycleModes == nil &&
 		!utilfeature.DefaultFeatureGate.Enabled(features.CSIInlineVolume) {
-		newCSIDriver := obj.(*storage.CSIDriver)
 		newCSIDriver.Spec.VolumeLifecycleModes = nil
 	}
-	if old.(*storage.CSIDriver).Spec.FSGroupPolicy == nil &&
+	if oldCSIDriver.Spec.FSGroupPolicy == nil &&
 		!utilfeature.DefaultFeatureGate.Enabled(features.CSIVolumeFSGroupPolicy) {
-		newCSIDriver := obj.(*storage.CSIDriver)
 		newCSIDriver.Spec.FSGroupPolicy = nil
 	}
-	if old.(*storage.CSIDriver).Spec.TokenRequests == nil &&
+	if oldCSIDriver.Spec.TokenRequests == nil &&
 		!utilfeature.DefaultFeatureGate.Enabled(features.CSIServiceAccountToken) {
-		csiDriver := obj.(*storage.CSIDriver)
-		csiDriver.Spec.TokenRequests = nil
+		newCSIDriver.Spec.TokenRequests = nil
 	}
-
-	if old.(*storage.CSIDriver).Spec.RequiresRepublish == nil &&
+	if oldCSIDriver.Spec.RequiresRepublish == nil &&
 		!utilfeature.DefaultFeatureGate.Enabled(features.CSIServiceAccountToken) {
-		csiDriver := obj.(*storage.CSIDriver)
-		csiDriver.Spec.RequiresRepublish = nil
+		newCSIDriver.Spec.RequiresRepublish = nil
+	}
+
+	// Any changes to the mutable fields increment the generation number.
+	if !reflect.DeepEqual(oldCSIDriver.Spec.TokenRequests, newCSIDriver.Spec.TokenRequests) || !reflect.DeepEqual(oldCSIDriver.Spec.RequiresRepublish, newCSIDriver.Spec.RequiresRepublish) {
+		newCSIDriver.Generation = oldCSIDriver.Generation + 1
```

Comment:
I don't see this being done for any other mutable fields in this group and for other core types like Pod. Is this necessary?

Reply:
this is mentioned by the removed comment above. honestly i am not sure if this is necessary. there are kinds using `Generation` e.g. `networking/ingressclass/strategy.go`

Reply:
it seems an informational field and as long as it is not decrementing, then it is fine. i.e. unset or increment. 

@liggitt do you have more to add about how this field is used?

Reply:
Incrementing generation is typically done for spec changes on objects which are reconciled by controllers and have a `status.observedGeneration` field, but it's fine to do this here as well.